### PR TITLE
Section exercises numbering

### PR DESCRIPTION
**Pull request by PatBoj**
I've created numbered section exercises title using template `Section [#chapter].[#section] Exercises`. My fix affects every single book, so regression tests are needed.

## **In the end of section**
image

## **In the _Answer Key_**
image

### CODE REVIEW

File: recipes/books/_generator.scss [hunk 1]
```diff
@@ -428,6 +426,7 @@
 
   //After: composite page creation
   @include count_countChapters(chapter);
+  @include count_countSections(section);
```
Comment (Adrianltd) [inline recipes/books/_generator.scss]: duplicate mixins for counting chapter and section (line 353 and 354)

File: recipes/books/_generator.scss [hunk 2]
```diff
@@ -350,12 +350,12 @@
   }
 }
 :pass(8) {
+  @include count_countChapters(chapter);
```
Comment (helenemccarron) [inline recipes/books/_generator.scss]: I would test this with the continuous numbering, is this going to break it? Also, we should make sure this applies only to books that need them, currently, this applies to all books with wide implications to numbering and would require full regression testing on all books to ensure numbering isn't affected.

File: recipes/books/_generator.scss (hunk 1 above)
Comment (PatBoj) [reply recipes/books/_generator.scss]: Yes, but they are in different passes, they need to be duplicated. Otherwise numbering won't work.

**Comment by PatBoj** [comment]
cc @helenemccarron @Adrianltd I've answered for your concerns and change the code a little bit.

File: recipes/books/_generator.scss [hunk 3]
```diff
@@ -350,12 +350,14 @@
   }
 }
 :pass(8) {
+  @if feature-enabled(eos-custom-titles) {
```
Comment (helenemccarron) [inline recipes/books/_generator.scss]: I was thinking something along the lines of `@if $Config_SectionCompositePages != ()` which checks if we are using EOS or not, do you think that would work? The benefits would be we wouldn't have to manage a feature flag.

**Comment by PatBoj** [comment]
@helenemccarron I believe I finally finished this issue :D. @Adrianltd could you take a look as well.

**Comment by Adrianltd** [comment]
I understand that `@include count_countChapters(chapter);` together with `@include count_countSections(section);` are replicated in two different passes for some reason, but they would apply always anyway because of the same conditional. So they apply first in the beginning on pass 8 and then somewhere in pass 9. Is this behaviour required @helenemccarron?

**Comment by helenemccarron** [comment]
based on @PatBoj, they are both required for his fix to work. I am not sure what happens when you remove one or the other, so i can't answer without testing both cases (with and without the duplicate).

**Pull request merged**